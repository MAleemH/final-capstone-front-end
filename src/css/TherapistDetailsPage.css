* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.list_style_none {
  list-style: none;
}

.td_none {
  text-decoration: none;
}


.therapy_details_main {
  display: flex;
  justify-content: space-between;
  align-items: start;
}

.therapy_details_nav {
  width: 20%;
  margin: auto;
  height: 100vh;
  position: sticky;
  top: 0;
  bottom: 0;
}

.therapy_details_content {
  padding: 15px 0;
  width: 80%;
  margin: auto;
}
