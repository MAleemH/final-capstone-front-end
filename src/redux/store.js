import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
