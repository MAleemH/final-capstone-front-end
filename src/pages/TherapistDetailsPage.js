import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { fetchReserves } from '../redux/Reserve/reserveSlice';
import Navigation from '../components/Navigation';
import '../css/TherapistDetails.css';
import therapistImg from '../img/therapist.jpg';
import editImg from '../img/edit.png';
import trashImg from '../img/trash.png';

function TherapistDetailsPage() {
  const dispatch = useDispatch();
  //  const myReserves = useSelector((state) => state.reserve.reserves);
  //  const { id } = useParams();
  //  const myReserve = (myReserves.filter((myReserve) => myReserve.id === id));

  useEffect(() => {
    let active = true;
    (async () => {
      if (active) {
        dispatch(fetchReserves());
      }
    })();
    return () => {
      active = false;
    };
  }, [dispatch]);

  return (
    <main className="therapy_details_main">

      <section className="therapy_details_nav">
        <Navigation />
      </section>

      <section className="therapy_details_content">
        <article>

          <div className="therapy_details_figure_box">
            <img src={therapistImg} alt="" />
          </div>

          <div className="therapist_info_box">
            <div className="therapist_info_contact_name">
              <h1>MARCUS DAVID</h1>
              <p>Lorem ipsum dolor sit.</p>
            </div>

            <ul className="list_style_none">

              <li>
                <p>RATING</p>
                <p>4</p>
              </li>
              <li>
                <p>SPECIALTY</p>
                <p>CARDIOVASCULAR</p>
              </li>
              <li>
                <p>LOCATION</p>
                <p>LAGOS</p>
              </li>
              <li>
                <p>EMAIL</p>
                <p>johndoe@yahoo.com</p>
              </li>
              <li>
                <p>AVAILABILITY</p>
                <p>TRUE</p>
              </li>
              <li>
                <p>PHONE</p>
                <p>07054245454</p>
              </li>
            </ul>

            <div className="therapist_info_action">
              <button aria-label="Book" type="button"><Link className="td_none" to="/book">Reserve</Link></button>
              <button aria-label="Edit" type="button"><img src={editImg} alt="" /></button>
              <button aria-label="Trash" type="button"><img src={trashImg} alt="" /></button>
            </div>

          </div>

        </article>
      </section>

    </main>
  );
}

export default TherapistDetailsPage;
