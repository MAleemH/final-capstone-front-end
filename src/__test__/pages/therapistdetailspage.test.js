import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import TherapistDetailsPage from '../../pages/TherapistDetailsPage';

it('Details page renders correctly', () => {
  const mysingleTherapist = store.getState().therapy.singleTherapist;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <TherapistDetailsPage mysingleTherapist={mysingleTherapist} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
