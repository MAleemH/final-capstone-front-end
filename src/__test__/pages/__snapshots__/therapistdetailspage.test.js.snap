// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page renders correctly 1`] = `
<main
  className="therapy_details_main"
>
  <section
    className="therapy_details_nav"
  >
    <main>
      <section
        className="home_nav_section"
      >
        <header
          className="home_logo"
        >
          <figure>
            <a
              className="td_none"
              href="/homepage"
              onClick={[Function]}
            >
              <img
                alt=""
                src="logo.png"
              />
            </a>
          </figure>
        </header>
        <nav
          className="list_style_none home_nav"
        >
          <li>
             
            <a
              activeClassName="active"
              className="td_none"
              href="/homepage"
              onClick={[Function]}
            >
              HOME
            </a>
          </li>
          <li>
             
            <a
              activeClassName="active"
              className="td_none"
              href="/appointments"
              onClick={[Function]}
            >
              APPOINTMENTS
            </a>
          </li>
          <li>
             
            <a
              activeClassName="active"
              className="td_none"
              href="/book"
              onClick={[Function]}
            >
              NEW APPOINTMENT
            </a>
          </li>
          <li>
            <button
              className="logout_btn"
              onClick={[Function]}
              type="button"
            >
              LOGOUT
            </button>
          </li>
        </nav>
        <footer
          className="home_nav_footer"
        >
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="facebook.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="twitter.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="linkedin.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="instagram.png"
              />
            </a>
          </figure>
          <figure
            className="nav_social_icons"
          >
            <a
              className="td_none"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                src="snapchat.png"
              />
            </a>
          </figure>
        </footer>
      </section>
      <section
        className="home_nav_section_mobile"
      >
        <header
          className="home_logo margin_left1"
        >
          <button
            aria-label="button"
            onClick={[Function]}
            type="button"
          >
            <img
              alt=""
              src="menu.png"
            />
          </button>
        </header>
      </section>
    </main>
  </section>
  <section
    className="therapy_details_content"
  />
</main>
`;
